flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -354,8 +355,54 @@ protected ClusterClient<String> getClusterClient(Configuration config) throws Ex
     }
 
     public void cancelJob(FlinkDeployment deployment, UpgradeMode upgradeMode) throws Exception {
-        var conf = configManager.getObserveConfig(deployment);
-        var deploymentStatus = deployment.getStatus();
+        Optional<String> savepointOpt = Optional.empty();
+
+        Configuration conf = null;
+        FlinkDeploymentStatus deploymentStatus = deployment.getStatus();
+        try {
+            conf = configManager.getObserveConfig(deployment);
+            savepointOpt = cancelJob(deployment, upgradeMode, conf, deploymentStatus);
+        } catch (LastReconciledSpecException e) {
+            LOG.warn("Last reconciled spec is null", e);
+        }
+
+        switch (upgradeMode) {
```

Comment:
Could you please not refactor the method? Seems like a lot of unnecessary changes, it’s hard to see what actually changed in the logic

Reply:
The method would be enormous. It seemed like a good practice to me to separate the behavior of deleting the cluster using the flink client and, on the other hand, deleting the kubernetes deployment. At first it is more complicated but then it becomes easier to read